chain-test/src/test/scala/org/bitcoins/chain/util/ChainUnitTest.scala

```diff
@@ -48,24 +55,30 @@ trait ChainUnitTest
     * @param test
     * @return
     */
-  def withBlockHeaderDAO(
-      test: BlockHeaderDAO => Future[Assertion]): Future[Assertion] = {
+  def withBlockHeaderDAO(test: OneArgAsyncTest): FutureOutcome = {
 
     val genesisHeaderF = setupHeaderTableWithGenesisHeader()
 
     val blockHeaderDAOF = genesisHeaderF.map(_ => blockHeaderDAO)
-    val testExecutionF = blockHeaderDAOF.flatMap(test(_))
+    val testExecutionF = blockHeaderDAOF.flatMap {
+      case dao: FixtureType => test(dao).toFuture
```

Comment:
this is really frustrating that this doesn't just work :tm:

Reply:
So basically for every fixture type we will need something like this, and this PR seems to add more LOC. I'm skeptical